internal/webhooks/vspherevm_webhook.go

```diff
@@ -63,48 +77,46 @@ func (r *VSphereVM) ValidateCreate() (admission.Warnings, error) {
 		}
 	}
 
-	if r.Spec.OS == Windows && len(r.Name) > 15 {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("name"), r.Name, "name has to be less than 16 characters for Windows VM"))
+	if objValue.Spec.OS == infrav1.Windows && len(objValue.Name) > 15 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("name"), objValue.Name, "name has to be less than 16 characters for Windows VM"))
 	}
 	if spec.GuestSoftPowerOffTimeout != nil {
-		if spec.PowerOffMode != VirtualMachinePowerOpModeTrySoft {
+		if spec.PowerOffMode != infrav1.VirtualMachinePowerOpModeTrySoft {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "guestSoftPowerOffTimeout"), spec.GuestSoftPowerOffTimeout, "should not be set in templates unless the powerOffMode is trySoft"))
 		}
 		if spec.GuestSoftPowerOffTimeout.Duration <= 0 {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "guestSoftPowerOffTimeout"), spec.GuestSoftPowerOffTimeout, "should be greater than 0"))
 		}
 	}
-	return nil, aggregateObjErrors(r.GroupVersionKind().GroupKind(), r.Name, allErrs)
+	return nil, aggregateObjErrors(objValue.GroupVersionKind().GroupKind(), objValue.Name, allErrs)
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type.
 //
 //nolint:forcetypeassert
-func (r *VSphereVM) ValidateUpdate(old runtime.Object) (admission.Warnings, error) {
+func (webhook *VSphereVMWebhook) ValidateUpdate(_ context.Context, oldRaw runtime.Object, newRaw runtime.Object) (admission.Warnings, error) {
 	var allErrs field.ErrorList
-	if r.Spec.GuestSoftPowerOffTimeout != nil {
-		if r.Spec.PowerOffMode != VirtualMachinePowerOpModeTrySoft {
-			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "guestSoftPowerOffTimeout"), r.Spec.GuestSoftPowerOffTimeout, "should not be set in templates unless the powerOffMode is trySoft"))
+
+	oldTyped := oldRaw.(*infrav1.VSphereVM) //nolint:forcetypeassert
```

Comment:
When we are already here: I'd prefer to have error handling instead of nolint here so we do the same as CAPI.

```suggestion
	oldTyped, ok := oldRaw.(*infrav1.VSphereVM) //nolint:forcetypeassert
	if !ok {
		return nil, apierrors.NewBadRequest(fmt.Sprintf("expected a VSphereVM but got a %T", oldRaw))
	}
```

Also for other occurencies. We could also then remove the above `//nolint:forcetypeassert` too.

Reply:
Fixed.